rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -596,12 +604,27 @@ NodeParameters::set_parameters_atomically(const std::vector<rclcpp::Parameter> &
 
   rcl_interfaces::msg::SetParametersResult result;
 
+  // check for any pre-set parameter registered callbacks
```

Comment:
Is there an issue if we instead moved the pre/post callbacks into `__set_parameters_atomically_common`?
It seems more intuitive to me to process those callbacks there, since technically that's where parameters are ultimately being set. As is currently implemented, pre/post callbacks won't be called when parameters are declared, but the validation callbacks do get called. Is there a reason for this asymmetry?